### DISCUSSION TITLE
Parse text tags

### DISCUSSION THREAD

**Discussion by tmjoen** [ueberdosis/tiptap#1569] [General Questions]
Hi! I'm struggling a bit with an extension that should format liquid type variables `{{ my_var }}`.
I have input and paste regexes that work:

```
export const inputRegex = /(?:^|\s)((?:{{)((?:[^}]+))(?:}}))$/gm
export const pasteRegex = /(?:^|\s)((?:{{)((?:[^}]+))(?:}}))/gm
```

but it won't format my initial content, so if I set my content in the `new Editor({...})` call, the variable is not formatted. I suppose I should handle this in `parseHTML`, but I'm not sure if it is possible to handle just text?

**Comment by hanspagel** [comment]
It’s not! inputRules are for typing, pasteRules for pasting and parseHTML to restore the document from a given HTML string.

**Reply by mshwery** [reply]
If you don't keep the html state around, what's the recommendation for rehydrating the editor with plain text and having it detect matching nodes?

**Reply by hanspagel** [reply]
Not sure if I understand your question correctly, but I’d recommend to keep the HTML/JSON around.

**Reply by mshwery** [reply]
I suppose my situation is somewhere in between that of a syntax highlighter (with much richer/extensible interaction handling that bubbles outside the editor) and that of a rich text editor. 

The shape of the persisted JSON data is different from the prosemirror html. Ideally I could decouple the shape of the data from the resulting prosemirror representation and define how that translates to prosemirror entities. Does that help shed some light on where my question is coming from?

**Reply by hanspagel** [reply]
Unfortunately not. You’re listing a few solutions you’d like to have, but I don’t know what problem you’re trying to solve.
